workspace-manager-clienttests/src/main/java/scripts/uploadscripts/SummariesToBigQuery.java

```diff
@@ -90,13 +93,16 @@ public void uploadResults(
         // insert into testScriptResults
         tableId = TableId.of(datasetName, testScriptResultsTableName);
         InsertAllRequest.Builder insertRequestBuilder = InsertAllRequest.newBuilder(tableId);
-        for (int ctr = 0; ctr < renderedTestConfiguration.testScripts.size(); ctr++) {
-            TestScriptSpecification testScriptSpecification =
-                    renderedTestConfiguration.testScripts.get(ctr);
-            TestScriptResult.TestScriptResultSummary testScriptResult =
-                    testRunSummary.testScriptResultSummaries.get(ctr);
-            insertRequestBuilder.addRow(
-                    buildTestScriptResultsRow(testScriptSpecification, testScriptResult));
+        Map<String, TestScriptResult.TestScriptResultSummary> testScriptResultSummaries =
+                new ConcurrentHashMap<String, TestScriptResult.TestScriptResultSummary>();
+        testRunSummary.testScriptResultSummaries.stream()
+                .forEach(runSummary -> testScriptResultSummaries.put(runSummary.testScriptDescription, runSummary));
+
+        for (TestScriptSpecification testScriptSpecification : renderedTestConfiguration.testScripts) {
+            if (testScriptResultSummaries.containsKey(testScriptSpecification.name)) {
+                insertRequestBuilder.addRow(
+                        buildTestScriptResultsRow(testScriptSpecification, testScriptResultSummaries.get(testScriptSpecification.name)));
+            }
```

Comment:
for this PR or as an enhancement (your choice), add an `else` clause here to handle the case where the `testScriptSpecification.name` is not found in the `testScriptResultSummaries`. It's worth at least logging this case so we know something went awry; you could also potentially insert some kind of "summary missing" row into BigQuery.

Reply:
Added `logger.debug` to track test names that are missing from test run summaries.